src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -125,5 +126,43 @@ CHIP_ERROR OTAImageProcessorImpl::ProcessHeader(ByteSpan & block)
     return CHIP_NO_ERROR;
 }
 
+// external flash optimization
+void ExtFlashHandler::DoAction(Action action)
+{
+    // introduce the QSPI driver into idle power mode
+    static constexpr auto sFlashDevice{ DT_LABEL(DT_INST(0, nordic_qspi_nor)) };
```

Comment:
Shouldn't we make it dependent on CONFIG_NORDIC_QSPI_NOR value? I think we shouldn't assume using QSPI NOR in generic OTA file, as some customers use SPI and it will not work for them.

Reply:
> In regard to other customers: that is actually why I have chosen more modular approach

I think that is very good, but I just wonder if the compiler will cut-off this method in the optimization process when `CONFIG_NORDIC_QSPI_NOR` will be false or not. I don't have anything against the abstraction, but just wondering about flash usage. :smile:

Reply:
I would assume that `gcc` is pretty clever about dead code and the whole unused implementation of `ExtFlashHandler` and  `OTAImageProcessorImplPMDevice` shall be snipped out from the resulting binary. But to be sure I have just built the `lock-app` with:

`west build -b nrf5340dk_nrf5340_cpuapp -- -DOVERLAY_CONFIG=overlay-low_power.conf` and `CONFIG_NORDIC_QSPI_NOR=n`

for the unused code tagged with and without CONFIG_NORDIC_QSPI_NOR ifdefs. There is no difference in resulting FLASH occupancy - 592400 bytes (61.32%) in both cases. So I hope we can keep the existing changes, as my initial approach was to have as less ifdefs as possible (actually have it only once) :)

Reply:
Ok, good to know, resolving that one then :)